net/local/local_sendmsg.c

```diff
@@ -334,6 +334,16 @@ static ssize_t local_sendto(FAR struct socket *psock,
       goto errout_with_halfduplex;
     }
 
+  /* Make sure that dgram is sent safely */
+
+  ret = nxsem_wait_uninterruptible(&conn->lc_sendsem);
+  if (ret < 0)
+    {
+      /* May fail because the task was canceled. */
+
+      return ret;
```

Comment:
goto errout_with_sender and close lc_outfile

Reply:
ok